src/mono/wasm/runtime/loader/assets.ts

```diff
@@ -422,9 +552,73 @@ function download_resource(request: ResourceRequest): LoadingResource {
     }
 }
 
+async function download_resource_with_cache(request: ResourceRequest): Promise<Response> {
+    let response = await findCachedResponse(request);
+    if (!response) {
+        response = await fetchResource(request);
+        addCachedReponse(request, response);
+    }
+
+    return response;
+}
+
+const monoToBlazorAssetTypeMap: { [key: string]: WebAssemblyBootResourceType | undefined } = {
+    "resource": "assembly",
+    "assembly": "assembly",
+    "pdb": "pdb",
+    "icu": "globalization",
+    "vfs": "configuration",
+    "dotnetwasm": "dotnetwasm",
+};
+
+const credentialsIncludeAssetBehaviors: AssetBehaviors[] = ["vfs"]; // Previously only configuration
+
+function fetchResource(request: ResourceRequest): Promise<Response> {
+    // Allow developers to override how the resource is loaded
+    let url = request.resolvedUrl!;
+    if (loaderHelpers.loadBootResource) {
+        const resourceType = monoToBlazorAssetTypeMap[request.behavior];
+        if (resourceType) {
```

Comment:
please comment the `else` fallthru

Reply:
We will call with `AssetBehaviors` and if we don't get result, we try with mapped `WebAssemblyBootResourceType`